controllers/datadogagent/utils.go

```diff
@@ -1694,6 +1699,41 @@ func isKSMCoreEnabled(dda *datadoghqv1alpha1.DatadogAgent) bool {
 	return false
 }
 
+func prometheusScrapeEnvVars(logger logr.Logger, dda *datadoghqv1alpha1.DatadogAgent) []corev1.EnvVar {
+	envVars := []corev1.EnvVar{}
+	if dda.Spec.Features == nil || dda.Spec.Features.PrometheusScrape == nil {
+		return envVars
+	}
+
+	if datadoghqv1alpha1.BoolValue(dda.Spec.Features.PrometheusScrape.Enabled) {
+		envVars = append(envVars, corev1.EnvVar{
+			Name:  datadoghqv1alpha1.DDPrometheusScrapeEnabled,
+			Value: datadoghqv1alpha1.BoolToString(dda.Spec.Features.PrometheusScrape.Enabled),
+		})
+
+		if dda.Spec.Features.PrometheusScrape.ServiceEndpoints != nil {
+			envVars = append(envVars, corev1.EnvVar{
+				Name:  datadoghqv1alpha1.DDPrometheusScrapeServiceEndpoints,
+				Value: datadoghqv1alpha1.BoolToString(dda.Spec.Features.PrometheusScrape.ServiceEndpoints),
+			})
+		}
+
+		if dda.Spec.Features.PrometheusScrape.AdditionalConfigs != nil {
+			jsonValue, err := yaml.YAMLToJSON([]byte(*dda.Spec.Features.PrometheusScrape.AdditionalConfigs))
+			if err != nil {
+				logger.Error(err, "Invalid additional prometheus config, ignoring it")
```

Comment:
What do you think about returning an error from here on invalid YAML and converting that into a reconciliation error visible on the `DatadogAgent` resource. This seems like a good case for letting the user know their config is invalid, otherwise they'd have to track down in logs, which may not be trivial for some users.

Reply:
I thought about it actually, and I choose to log the error and unblock the reconciling for 2 reasons
- The config is not for a critical feature in the agent, I found it too agressive to allow it to block the reconciliation
- The operator doesn't provide an agent config validation (example: the agent env vars, `confd` files, `datadog.yaml`) all can be passed via the CRD and the agent will then validate them. That being said, returning the error here is not aligned with what the operator does elsewhere.

Reply:
Interesting, ok then, I do wish we did more validation though of those config settings as well. Even if we don't want to bock we could add a "misconfiguration detected" condition. Something to think about later though.